tabrepo/utils/huggingfacehub_utils.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+
+from huggingface_hub import HfApi
+from tqdm import tqdm
+
+
+def upload_hugging_face(version: str, repo_id: str, override_existing_files: bool = True):
+    """
+    Uploads tabrepo data to Hugging Face repository.
+    You should set your env variable HF_TOKEN and ask write access to tabrepo before using the script.
+
+    Args:
+        version (str): The version of the data to be uploaded, the folder data/results/{version}/ should
+        be present and should contain baselines.parquet, configs.parquet and model_predictions/ folder
+        repo_id (str): The ID of the Hugging Face repository.
+        override_existing_files (bool): Whether to re-upload files if they are already found in HuggingFace.
+    Returns:
+        None
+    """
+    commit_message = f"Upload tabrepo new version"
+    root = Path(__file__).parent.parent.parent / f"data/results/{version}/"
+
+    for filename in ["baselines.parquet", "configs.parquet", "model_predictions"]:
+        assert (root / filename).exists(), f"Expected to found {filename} but could not be found in {root / filename}."
+    api = HfApi()
+    for filename in ["baselines.parquet", "configs.parquet"]:
+        path_in_repo = str(Path(version) / filename)
+        if api.file_exists(repo_id=repo_id, filename=path_in_repo, token=os.getenv("HF_TOKEN"), repo_type="dataset") and not override_existing_files:
+            print(f"Skipping {path_in_repo} which already exists in the repo.")
+            continue
+
+        api.upload_file(
+            path_or_fileobj=root / filename,
+            path_in_repo=path_in_repo,
+            repo_id=repo_id,
+            repo_type="dataset",
+            commit_message=commit_message,
+            token=os.getenv("HF_TOKEN"),
+        )
+    files = list(sorted(Path(root / "model_predictions").glob("*")))
+    for dataset_path in tqdm(files):
+        print(dataset_path)
+        try:
+            path_in_repo = str(Path(version) / "model_predictions" / dataset_path.name)
+            # ideally, we would just check if the folder exists but it is not possible AFAIK, we could alternatively
+            # upload per file but it would create a lot of different commits.
+            if api.file_exists(repo_id=repo_id, filename=str(Path(path_in_repo) / "0" / "metadata.json"), token=os.getenv("HF_TOKEN"),
+                               repo_type="dataset") and not override_existing_files:
+                print(f"Skipping {path_in_repo} which already exists in the repo.")
+                continue
+            api.upload_folder(
+                folder_path=dataset_path,
+                path_in_repo=path_in_repo,
+                repo_id=repo_id,
+                repo_type="dataset",
+                ignore_patterns="*DS_Store",
+                commit_message=f"Upload tabrepo new version {dataset_path.name}",
+                token=os.getenv("HF_TOKEN"),
+            )
+        except Exception as e:
+            print(str(e))
```

Comment:
Why not raise an exception?

Reply:
To allow other datasets to upload. I will added a flag to control this.